packages/headless/src/features/generated-answer/generated-answer-state.ts

```diff
@@ -1,12 +1,37 @@
 import {GeneratedAnswerCitation} from '../../api/generated-answer/generated-answer-event-payload';
 
 export interface GeneratedAnswerState {
+  /**
+   * Determines if the generated answer is loading.
+   */
   isLoading: boolean;
+  /**
+   * Determines if the generated answer is streaming.
+   */
   isStreaming: boolean;
+  /**
+   * The generated answer.
+   */
   answer?: string;
+  /**
+   * The citations used to generate the answer.
```

Comment:
This is slightly incorrect. The citations are all the document snippets retrieved and sent to the generative model but it is not guaranteed that they have all been used by the model.

Reply:
"The document snippets retrieved to generate the answer." ?